tools/libcloud/gcp.py

```diff
@@ -175,7 +175,8 @@ def get_boot_node(gce):
     nodes = gce.list_nodes()
     for node in nodes:
         if node.name.startswith("core-boot-"):
-            gce.wait_until_running([node], 5, 600) # 5 second loops | 10 minute timeout
+            while(not node.public_ips):
```

Comment:
nope, you need to refresh the `node` metadata or you'll be stuck in this while loop forever:

```python
while not node.public_ips:
    logging.info("Waiting for %s to get its public IP ...", node.name)
    time.sleep(5)
    node = gce.ex_get_node(node.name)
 ```